File: Projet_final/JavaProject/src/main/java/model/Formateur.java
package model;

import java.util.Date;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("Formateur")
public class Formateur extends User {
	@Column(name = "Formateur_disponibilites")
	@JsonView(JsonViews.Common.class)
	private Date[] disponibilites;
	@OneToMany(mappedBy = "key.formateur")
	@JsonView(JsonViews.User.class)
	private Set<Competence> competences;
	@OneToMany(mappedBy = "formateur")
	@JsonView(JsonViews.User.class)
	private Set<Module> modules;
	@OneToMany(mappedBy = "referent")
	@JsonView(JsonViews.User.class)
	private Set<Cursus> setCursus;

	public Date[] getDisponibilite() {
		return disponibilites;
	}

	public void setDisponibilite(Date[] disponibilite) {
		this.disponibilites = disponibilite;
	}

	public Set<Competence> getCompetences() {
		return competences;
	}

	public void setCompetences(Set<Competence> competences) {
		this.competences = competences;
	}

	public Set<Module> getModules() {
		return modules;
	}

	public void setModules(Set<Module> modules) {
		this.modules = modules;
	}

	public Set<Cursus> getCursus() {
		return setCursus;
	}

	public void setCursus(Set<Cursus> setCursus) {
		this.setCursus = setCursus;
	}
}

File: Projet_final/angularBT/src/app/model/formateur.ts
import { Competences } from './competences';
import { Cursus } from './cursus';
import { Module } from './module';
import { User } from './user';
export class Formateur extends User{
  constructor(private _disponnibilite: Date[] = null, private _competences: Competences=null, private _module: Module=null, private _cursus: Cursus=null) {
    super();
  }
  
   public get disponnibilite(): Date[] {
    return this._disponnibilite;
  }
  public set disponnibilite(_disponnibilite: Date[]) {
    this._disponnibilite = _disponnibilite;
  }
 public get competences(): Competences {
    return this._competences;
  }
  public set competences(_competences: Competences) {
    this._competences = _competences;
  }
   public get module():  Module {
    return this._module;
  }
   public set module(_module: Module) {
    this._module = _module;
  } 
   public get cursus():  Cursus {
    return this._cursus;
  }
}